### PR TITLE
Fix wrong declairingType in groovy properties when get genericReturnType.

### PR DESCRIPTION
**Pull request by altro3**
Fix wrong declairingType in groovy properties when get genericReturnType.

See https://github.com/micronaut-projects/micronaut-openapi/issues/670

### CODE REVIEW

File: inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/GroovyClassElement.java [hunk 1]
```diff
@@ -851,11 +851,22 @@ private List<GroovyPropertyElement> getPropertyNodes() {
...
+                        propDeclaringClass,
+                        AstAnnotationUtils.getAnnotationMetadata(sourceUnit, compilationUnit, propDeclaringClass)
+                    );
+                }
+
                 final AnnotationMetadata annotationMetadata =
                         AstAnnotationUtils.getAnnotationMetadata(sourceUnit, compilationUnit, propertyNode.getField());
                 GroovyPropertyElement groovyPropertyElement = new GroovyPropertyElement(
                         visitorContext,
-                        this,
```
Comment (dstepanov) [inline inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/GroovyClassElement.java]: Please keep the changes as before and introduce the same code as in `GroovyMethodElement` to  `GroovyPropertyElement`.

**Review by dstepanov — CHANGES_REQUESTED** [review]

File: inject-groovy/src/test/groovy/io/micronaut/inject/visitor/ClassElementSpec.groovy [hunk 2]
```diff
@@ -44,6 +47,77 @@ class ClassElementSpec extends AbstractBeanDefinitionSpec {
...
+        def method = AllElementsVisitor.VISITED_METHOD_ELEMENTS[0]
+        def type = method.genericReturnType;
+        List<PropertyElement> beanProperties = type.getBeanProperties().stream().filter(p -> !"groovy.lang.MetaClass".equals(p.getType().getName())).collect(Collectors.toList());
+
+        List<TypedElement> childClassFields = new ArrayList<>();
+        for (TypedElement publicField : beanProperties) {
+            if (publicField instanceof MemberElement) {
+
+                MemberElement memberEl = (MemberElement) publicField;
+                if (memberEl.getDeclaringType().getType().getName() == type.getName()) {
```
Comment (dstepanov) [inline inject-groovy/src/test/groovy/io/micronaut/inject/visitor/ClassElementSpec.groovy:111]: Please also test that owner element is the correct one

File: inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/GroovyClassElement.java [hunk 3]
```diff
@@ -851,11 +851,22 @@ private List<GroovyPropertyElement> getPropertyNodes() {
...
+                    declaringEl = new GroovyClassElement(
+                        visitorContext,
+                        propDeclaringClass,
+                        AstAnnotationUtils.getAnnotationMetadata(sourceUnit, compilationUnit, propDeclaringClass)
+                    );
+                }
+
                 final AnnotationMetadata annotationMetadata =
                         AstAnnotationUtils.getAnnotationMetadata(sourceUnit, compilationUnit, propertyNode.getField());
                 GroovyPropertyElement groovyPropertyElement = new GroovyPropertyElement(
```
Comment (dstepanov) [inline inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/GroovyClassElement.java]: FYI most of the code in elements contains `declaredType` but that's actually owned type. I'm working on a PR that changes it.

**Review by dstepanov — CHANGES_REQUESTED** [review]

**Comment by altro3** [comment]
@dstepanov done

**Review by dstepanov — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

**Pull request merged**